### PR TITLE
Apply ruff/flake8-implicit-str-concat rule ISC001

### LINKED ISSUE #1402: Add ruff/flake8-implicit-str-concat rules (ISC)
**How would this feature be useful?**

[`ISC001`](https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/) detects implicitly concatenated strings on a single line. They are usually left-overs of `black` or `ruff format` runs.

**Describe the solution you'd like**

Fix issues reported by `ruff check --extend-select ISC001` and add `ISC` to `pyproject.toml`.

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
- [x] I have added a news fragment under `changelog.d/` (if the patch affects the end users)

## Summary of changes

Apply ruff/flake8-implicit-str-concat rule `ISC001`

Fixes #1402 .

## Test plan

Tested by running

```console
$ ruff check --extend-select ISC
All checks passed!
$ 
```

### CODE REVIEW

File: scripts/update_package_cache.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def update_test_packages_cache(package_list_dir_path: Path, pipx_package_cache_p
 
     if not platform_package_list_path.exists():
         print(
-            f"WARNING.  File {str(platform_package_list_path)}\n" "    does not exist.  Creating now...",
+            f"WARNING.  File {str(platform_package_list_path)}\n    does not exist.  Creating now...",
```
Comment (DimitriPapadopoulos) [inline scripts/update_package_cache.py:66]: Not sure what to do here. In #1118 / 7184de2 the `black` run folded this multiline string:
```python
        print(
            f"WARNING.  File {str(platform_package_list_path)}\n"
            "    does not exist.  Creating now...",
            file=sys.stderr,
        )
```
into:
```python
        print(
            f"WARNING. File {str(platform_package_list_path)}\n" " does not exist. Creating now...",
            file=sys.stderr,
        )
``` 

Reverting is not an option, as the next run of `black` will overwrite once again.

Perhaps use multiline strings?
```python
        print(
            f"""\
WARNING.  File {str(platform_package_list_path)}"
    does not exist.  Creating now..."'',
            file=sys.stderr,
        )
```

**Review by gaborbernat — DISMISSED** [review]

**Comment by DimitriPapadopoulos** [comment]
I had to rebase.

**Review by gaborbernat — APPROVED** [review]

**Review by gaborbernat — DISMISSED** [review]

File: src/pipx/main.py [hunk 2]
```diff
@@ -806,7 +806,7 @@ def _add_runpip(subparsers, venv_completer: VenvCompleter, shared_parser: argpar
 def _add_ensurepath(subparsers: argparse._SubParsersAction, shared_parser: argparse.ArgumentParser) -> None:
     p = subparsers.add_parser(
         "ensurepath",
-        help=("Ensure directories necessary for pipx operation are in your " "PATH environment variable."),
+        help=("Ensure directories necessary for pipx operation are in your PATH environment variable."),
```
Comment (uranusjr) [inline src/pipx/main.py:809]:
```suggestion
        help="Ensure directories necessary for pipx operation are in your PATH environment variable.",
```

**Review by uranusjr — APPROVED** [review]

File: changelog.d/1402.misc.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Enable and apply ruff rule `ISC001`.
```
Comment (huxuan) [inline changelog.d/1402.misc.md:1]: Since this does not affect end users, there is no need to have a changelog.

Comment (DimitriPapadopoulos) [reply changelog.d/1402.misc.md]: I have removed `changelog.d/1402.misc.md`.

You might want to remove `changelog.d/1400.misc.md` as well.

**Review by huxuan — APPROVED** [review]

**Pull request merged**